changing ellipticity definition for the potential

sibirrer opened the pull request:
changing ellipticity definition for the potential in NFW_ELLIPSE, PJffeEllipse and SERSIC_ELLIPSE_POT models

mattgomer commented on lenstronomy/LensModel/Profiles/p_jaffe_ellipse.py:
```diff
@@ -79,8 +68,8 @@ def mass_3d_lens(self, r, sigma0, Ra, Rs, e1=0, e2=0):
         :param Rs:
-        :param q:
-        :param phi_G:
+        :param e1:
```
Are these e1 and e2 parameters even used in this function?

sibirrer replied:
it is added because the input function does come (potentially) with this parameter

mattgomer commented on lenstronomy/LensModel/Profiles/p_jaffe_ellipse.py:
```diff
@@ -25,14 +25,7 @@ def function(self, x, y, sigma0, Ra, Rs, e1, e2, center_x=0, center_y=0):
-        e = min(abs(1. - q), 0.99)
-        x_ = (cos_phi*x_shift+sin_phi*y_shift)*np.sqrt(1 - e)
-        y_ = (-sin_phi*x_shift+cos_phi*y_shift)*np.sqrt(1 + e)
+        x_, y_ = param_util.transform_e1e2_square_average(x, y, e1, e2, center_x, center_y)
```
I'm not very familiar with p_jaffe. Is ellipticity added in potential like the other cases? If it is added in mass, there may be an inconsistency.

sibirrer replied:
yes, it's in potential. I added a detailed description of the profile

mattgomer commented on lenstronomy/Util/param_util.py:
```diff
@@ -146,7 +146,19 @@ def transform_e1e2_square_average(x, y, e1, e2, center_x, center_y):
+    :param q:
+    :return:
+    """
+    e = (1 - q ** 2) / (1 + q ** 2)
```
Maybe the numerator needs and abs(). Not sure why abs(1-q) was used before. Are there ever cases with q>1?

sibirrer replied:
indeed, that can be done for more stability

sibirrer commented:
@mattgomer I updated lenstronomy and fixed some tests. The test coverage reduced because the removal of redundant code lines. Let me know if it is ok or not! Thank you!

mattgomer reviewed (approved):
Looks good, provided the hard-coded values in the test files are correct.
